File: leetcode/cpp/two_sum.cc
class Solution {
public:
  vector<int> twoSum(vector<int> &numbers, int target) {
    int size = numbers.size();
    unordered_map<int, int> hash;
    vector<int> ret;
    for (int i = 0; i < size; i++) {
      int diff = target - numbers[i];
      if (hash.find(diff) != hash.end()) {
	ret.push_back(hash[diff]);
	ret.push_back(i+1);
        return ret;
      }
      hash[numbers[i]] = i+1;
    }
    return ret;
  }
};

File: leetcode/python/two_sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        table = {}
        ret = []
        for i, num in enumerate(nums, 1):
            if target-num in table:
                ret.append(table[target-num])
                ret.append(i)
                break
            else:
                table[num] = i
        return ret

if __name__ == "__main__":
	sol = Solution()
	print sol.twoSum([3, 2, 4], 6)
